crates/holochain/src/conductor/conductor/tests/agent_key_revocation.rs

```diff
@@ -0,0 +1,219 @@
+    let result = conductor
+        .clone()
+        .revoke_agent_key_for_app(agent_key.clone(), app.installed_app_id().clone())
+        .await;
+    assert_matches!(result, Err(ConductorError::DpkiError(DpkiServiceError::DpkiAgentInvalid(invalid_key, _))) if invalid_key == agent_key);
+
+    // reading an entry should still succeed
+    let result: Option<Record> = conductor.call(&zome_1, read_fn_name, action_hash_1).await;
+    assert!(result.is_some());
+    let result: Option<Record> = conductor.call(&zome_2, read_fn_name, action_hash_2).await;
+    assert!(result.is_some());
+
+    // creating an entry should fail now for both cells
+    let result = conductor
+        .call_fallible::<_, ActionHash>(&zome_1, create_fn_name, ())
+        .await;
+    if let Err(ConductorApiError::CellError(CellError::WorkflowError(workflow_error))) = result {
+        assert_matches!(
+            *workflow_error,
+            WorkflowError::SysValidationError(SysValidationError::ValidationOutcome(ValidationOutcome::DpkiAgentInvalid(invalid_key, _))) if invalid_key == agent_key
+        );
+    } else {
+        panic!("different error than expected");
+    }
+    let result = conductor
+        .call_fallible::<_, ActionHash>(&zome_2, create_fn_name, ())
+        .await;
+    if let Err(ConductorApiError::CellError(CellError::WorkflowError(workflow_error))) = result {
+        assert_matches!(
+            *workflow_error,
+            WorkflowError::SysValidationError(SysValidationError::ValidationOutcome(ValidationOutcome::DpkiAgentInvalid(invalid_key, _))) if invalid_key == agent_key
+        );
+    } else {
+        panic!("different error than expected");
+    }
+
+    // last source chain action in both cells should be CloseChain
+    conductor
+        .get_or_create_authored_db(dna_file_1.dna_hash(), agent_key.clone())
+        .unwrap()
+        .test_read(|txn| {
+            txn.query_row(
+                "SELECT type FROM Action ORDER BY seq DESC LIMIT 1",
+                [],
+                |row| {
+                    let action_type = row.get::<_, String>("type").unwrap();
+                    assert_eq!(action_type, ActionType::CloseChain.to_string());
+                    Ok(())
+                },
+            )
+            .unwrap();
+        });
+    conductor
+        .get_or_create_authored_db(dna_file_2.dna_hash(), agent_key.clone())
+        .unwrap()
+        .test_read(|txn| {
+            txn.query_row(
+                "SELECT type FROM Action ORDER BY seq DESC LIMIT 1",
+                [],
+                |row| {
+                    let action_type = row.get::<_, String>("type").unwrap();
+                    assert_eq!(action_type, ActionType::CloseChain.to_string());
+                    Ok(())
+                },
+            )
+            .unwrap();
+        });
+
+    // cloning cells should fail for both cells
+    let result = conductor
+        .raw_handle()
```

Comment:
The method `SweetConductor::create_clone_cell` returns `ConductorApiError`, whereas `Conductor::create_clone_cell` returns `ConductorError`. Hence I use the raw handle here. Is there a shorter way/why is the sweet conductor using a different return type?

Reply:
That the sweet conductor uses a different type is probably a mistake, feel free to change that